pdf/model/textencoding/identity.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package textencoding
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/unidoc/unidoc/common"
+	"github.com/unidoc/unidoc/pdf/core"
+)
+
+// IdentityEncoder represents an 2-byte identity encoding
+type IdentityEncoder struct {
+	baseName string
+}
+
+// NewSimpleTextEncoder returns a IdentityEncoder based on predefined encoding `baseName` and
+// difference map `differences`.
+func NewIdentityTextEncoder(baseName string) IdentityEncoder {
+	return IdentityEncoder{baseName}
+}
+
+// String returns a string that describes `enc`.
+func (enc IdentityEncoder) String() string {
+	return enc.baseName
+}
+
+// Encode converts the Go unicode string `raw` to a PDF encoded string.
+func (enc IdentityEncoder) Encode(raw string) string {
+	// runes -> character codes -> bytes
+	var encoded bytes.Buffer
+	for _, r := range raw {
+		code, ok := enc.RuneToCharcode(r)
+		if !ok {
+			common.Log.Debug("Failed to map rune to charcode. rune=%+q", r)
+			continue
+		}
+
+		// Each entry represented by 2 bytes.
+		encoded.WriteByte(byte((code & 0xff00) >> 8))
+		encoded.WriteByte(byte(code & 0xff))
+	}
+	return encoded.String()
+}
+
+// CharcodeToGlyph returns the glyph name matching character code `code`.
+// The bool return flag is true if there was a match, and false otherwise.
+func (enc IdentityEncoder) CharcodeToGlyph(code uint16) (string, bool) {
+	r, found := enc.CharcodeToRune(code)
+	if found && r == 0x20 {
+		return "space", true
+	}
+
+	// Returns "uniXXXX" format where XXXX is the code in hex format.
+	glyph := fmt.Sprintf("uni%.4X", code)
+	return glyph, true
+}
+
+// Conversion between glyph name and character code.
```

Comment:
godoc.
// GlyphToCharcode converts a glyph to a charcode
Should we use the CharCode or uint16 here?  I remember there was a type definition of CharCode in a different location

Reply:
This is based on the `TextEncoder` interface.

`CharCode` was defined to work with any character code, 1, 2 or 4 byte.

I think the only place that 32 bit character codes occur are in ToUnicode maps, so it should be ok to leave `TextEncoder as `uint16`.

If you think using `CharCode` for all character codes makes the Go code easier read then I can change this. Readability is important.